applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c

```diff
@@ -181,42 +182,64 @@ int json_common_modem_dynamic_data_add(cJSON *parent,
 		goto exit;
 	}
 
-	/* Convert mccmnc to unsigned long integer. */
-	errno = 0;
-	mccmnc = strtoul(data->mccmnc, &end_ptr, 10);
-	if ((errno == ERANGE) || (*end_ptr != '\0')) {
-		LOG_ERR("MCCMNC string could not be converted.");
-		err = -ENOTEMPTY;
-		goto exit;
+	if (data->rsrp_valid) {
+		err = json_add_number(modem_val_obj, MODEM_RSRP, data->rsrp);
+		if (err) {
+			LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+			goto exit;
+		}
+		values_added = true;
 	}
 
-	err = json_add_number(modem_val_obj, MODEM_RSRP, data->rsrp);
-	if (err) {
-		LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
-		goto exit;
+	if (data->area_code_valid) {
+		err = json_add_number(modem_val_obj, MODEM_AREA_CODE, data->area);
+		if (err) {
+			LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+			goto exit;
+		}
+		values_added = true;
 	}
 
-	err = json_add_number(modem_val_obj, MODEM_AREA_CODE, data->area);
-	if (err) {
-		LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
-		goto exit;
+	if (data->mccmnc_valid) {
+		/* Convert mccmnc to unsigned long integer. */
+		errno = 0;
+		mccmnc = strtoul(data->mccmnc, &end_ptr, 10);
+		if ((errno == ERANGE) || (*end_ptr != '\0')) {
```

Comment:
A blank line before the `if` line would be nice. I thought checkpatch used to catch these before, but looks like it allows it, so it's just a suggestion from my side

Reply:
Agree. Will add.